fix(python): regex injection not working for concatenated strings

AbysmalBiscuit opened the pull request:
Hello,

Currently the Python regex injection query doesn't handle concatenated strings.
This PR fixes this by adding a new query to inject regex into concatenated strings.

Thank you for taking the time to review my PR. 😊

## Context

Splitting a regex pattern into multiple strings in Python can be beneficial for legibility, especially when you have to use very long regex expressions.

Example code:

```python
import re

re_test: re.Pattern[str] = re.compile(r"^(?P<year>\d{4}) (?P<day>\d) \w\s{,3}")
re_test: re.Pattern[str] = re.compile(
    r"^(?P<year>\d{4}) "
    r"(?P<day>\d) \w\s{,3}"
)
```

Both instances create equivalent `re.Pattern` objects.

## Before

Here is how it gets highlighted currently:
<img width="783" height="158" alt="image" src="https://github.com/user-attachments/assets/d24b1eb7-413b-4d08-b480-6e4eb2f6d624" />

The concatenated string in the second `re.compile` is not detected as regex.

<details><summary>Screenshot with Inspect tree view (without new query)</summary>
<p>

<img width="1691" height="1037" alt="image" src="https://github.com/user-attachments/assets/d36dca69-ebbd-4d2e-8255-518c283d7a76" />

</p>
</details> 

## After

Including the new additional query will take into account concatenated strings, so they get detected as regex:
<img width="770" height="156" alt="image" src="https://github.com/user-attachments/assets/5f2358b9-d048-47c6-9f57-af1f971894f3" />

The concatenated string in the second `re.compile` is now detected as regex.

<details><summary>Screenshot with Inspect tree view (with new query)</summary>
<p>

<img width="1695" height="1338" alt="image" src="https://github.com/user-attachments/assets/1463d86e-5a44-4aa6-b212-16f2dc9b7fa7" />

</p>
</details>

clason commented:
> A scoped/quantified injection would be more performant, but only supported on 0.11.3+

That's fine; the `main` branch is 0.11.4 only.

AbysmalBiscuit commented:
Hey @ribru17 and @clason, I updated the injection to use `@injection.combined` and updated the highlights query to set `@string.regexp` for concatenated strings.

I added the small example I gave as a test file.

AbysmalBiscuit commented:
@ribru17 thanks for the suggestions!

I think comments around the strings are indeed uncommon, but I can imagine them being useful.

I updated the injection to be scoped and to handle comments around the strings.
I also updated the highlights for `@string.regex` to be scoped as well.

ribru17 commented:
Any time! And sorry to do this to you, but would you mind splitting up the regular string and concatenated string patterns into two unique patterns? The singular pattern looks a bit convoluted. Also, imo, we would remove the preceding `(comment)*` and the start anchor in the patterns

Also, would you mind making the highlight pattern split, and also actually remove the quantifier? Those ones don't need to be quantified/scoped and it usually makes query iteration slightly slower

AbysmalBiscuit commented:
Hey, I had a look and to the best of my understanding it doesn't seem like `starlark` has a `re` module.

I updated the starlark injections to not inherit from python and added test files to make sure it works correctly.

I found [this third party project](https://github.com/magnetde/starlark-re) that adds it. If you think it's necessary I can add the non-concatenated regex injection query to the starlark injections.

ribru17 commented on runtime/queries/python/highlights.scm:
```diff
@@ -417,11 +417,22 @@
+        (string
+          (string_content) @string.regexp)
+        (comment)
+      ]))
```
```suggestion
      (string
        (string_content) @string.regexp)))
```

ribru17 reviewed (approved):
LGTM! One final nit:
